005-github-simplification.Rmd

```diff
@@ -0,0 +1,93 @@
+    This affects packages like dplyr where the underlying issue might need to be fixed in vctrs.
+
+## Solutions
+
+I think there are three possibilities:
+
+-   Move r-lib and tidymodels repos to **tidyverse org**.
+-   Move r-lib, tidyverse, and tidymodel repos to a **new org**.
+-   Move r-lib, tidyverse, and tidymodel to the **rstudio org**.
+
+These cases only affect GitHub organisations and in the short term all other websites would stay as is.
+This means that (e.g.) vroom would continue to live at vroom.r-lib.org.
+In the longer term, we'd probably make it vroom.{orgname}.org, but this would required some thought about how to manage the redirects, and whether it makes sense for (e.g.) parsnip to live at parsnip.{orgname}.org while the main tidymodels website continues to live at tidymodels.org.
+
+### Move to tidyverse org
+
+Pros:
+
+-   Moves smallest number of packages.
+-   Tidyverse has strongest brand identity.
+
+Cons:
+
+-   Makes it more complicated to explain exactly what the tidyverse is: it's a package, it's a set of packages loaded by `library(tidyverse)`, it's a set of packages installed by `install.packages("tidyverse")`, **and** it's a GitHub organisation.
+-   Binds all packages more closely to tidyverse. This might be less appealing for external developers and for other RStudio teams we might hope would also migrate to one org.
+-   No longer a one-to-one relationship between package homepage and GitHub organisation. (But could resolve in a second phase).
+-   Lose ability to bury less-important packages in r-lib; i.e. harder to highlight most important packages in tidyverse.
+
+### Create a new org
+
+Pros:
+
+-   May be easier to persuade other organisations to join if not "tidyverse" branded.
+
+Cons:
+
+-   Maximum amount of work moving packages.
+-   Will be hard to come up with good name.
+-   In the short term, increases numbers of orgs.
+
+### Move to rstudio org
+
+Pro:
+
+-   Greatest reduction in organizations.
+
+-   Easy story to tell (now that RStudio is a PBC, we have 0% concerns about it as long term steward of this code).
+
+Cons:
+
+-   Eliminates veneer of separation between RStudio and tidyverse, and makes it harder to invite/recruit non-employee authors and maintainers.
+-   Would fall under RStudio IT practices which are geared towards protecting private data. This will make it challenging to manage our own processes around teams.
+-   Would need very clear reassurance that this isn't RStudio "muscling in" on open source. Regardless of how we communicate this, a few people are likely to generate negative press.
+
+## Open questions
+
+-   GitHub's existing forwarding should handle most cases, but there a few cases where you'll need to update the urls in the GitHub repo. But are there any existing workflows that this would break?
+-   Should we consider the moving the websites as part of this process? If so, how do we manage the redirects? Will need to set up some external server.
```

Comment:
I think if repos are moving between orgs, it would be good to move the websites as part of the same phase, as opposed to a 2nd phase. It seems confusing to me to (e.g.) have `vroom.r-lib.org` as the URL but the repo is `tidyverse/vroom` or `rstudio/vroom`. I think it would be additionally confusing if this will change in a short-ish time span.

Reply:
agreed